.circleci/config.yml

```diff
@@ -376,6 +385,41 @@ commands:
             ICONS=pro yarn build-storybook
             ls -lRa ./storybook-build
 
+  e2e_build:
+    parameters:
+      test_runner:
+        type: enum
+        enum:
+          - testcafe
+          - playwright
+    steps:
+      - restore_repo
+      - setup_remote_docker:
+          version: *remote_docker_version
+          docker_layer_caching: false
+      - run:
+          name: Build and push E2E << parameters.test_runner >> image
+          working_directory: *workspace_compose
+          command: |
+            ecr-login
+            # The image is not expected to chage too often so we use :latest for all cache
+            E2E_IMAGE="307238562370.dkr.ecr.eu-west-1.amazonaws.com/evaka/e2e-<< parameters.test_runner >>:latest"
+            docker pull "$E2E_IMAGE"
+            docker build -t "$E2E_IMAGE" --cache-from "$E2E_IMAGE" -f "./e2e/<< parameters.test_runner >>.Dockerfile" ./e2e/
+            docker push "$E2E_IMAGE"
```

Comment:
One downside I can see to this, is that now this could not be run as-is on the theoretical "forks-only workflow without secrets" but not sure that it's a use case worth over-complicating this over (and rather just create some alternative solution).

Reply:
We can also build and save it on local too. Something like 
- restore from cache
- if cache
-- docker load
-- docker build with cache
- else
-- docker build without cache 
- store to cache

Reply:
Created flag to swich between these.